### PR TITLE
feat: enable export file

### PR DESCRIPTION
**Pull request by rdubigny**

### CODE REVIEW

**Comment by rdubigny** [comment]
[ETQ instructeur, je peux demander un Export des demandes Datapass](https://trello.com/c/nHYq676W/1339-etq-instructeur-je-peux-demander-un-export-des-demandes-datapass)

File: backend/app/controllers/enrollments_controller.rb [hunk 1]
```diff
@@ -1,4 +1,6 @@
 class EnrollmentsController < ApplicationController
+  include ActionController::MimeResponds
```
Comment (skelz0r) [inline backend/app/controllers/enrollments_controller.rb:2]: You don't actually need. This module includes `respond_to` method, which is convenient when you have to deal with multiple content type, which is not the case here.

File: backend/app/controllers/enrollments_controller.rb [hunk 2]
```diff
@@ -71,6 +73,15 @@ def index
       root: "enrollments"
   end
 
+  # GET /enrollments/export
+  def export
+    @enrollments = policy_scope(Enrollment)
+
+    respond_to do |format|
+      format.csv { send_data @enrollments.to_csv, filename: "export-datapass-#{Date.today}.csv" }
+    end
```
Comment (skelz0r) [inline backend/app/controllers/enrollments_controller.rb:82]:
```suggestion
    send_data @enrollments.to_csv, type: 'text/csv', disposition: 'attachment', filename: "export-datapass-#{Date.today}.csv"
```

1. Simplify according to my comment on the module 
2. Add type to force browser to download it (for old browser versions)
3. Specify to the browser to act as a downloadable link, which keeps the actual page open (more convenient)

File: backend/app/models/enrollment.rb [hunk 3]
```diff
@@ -169,6 +171,50 @@ def copy(current_user)
...
+      fondement_juridique_url
+      team_members_json
+      cgu_approved
+      dpo_is_informed
+      additional_content
+      linked_token_manager_id
+      previous_enrollment_id
+      copied_from_enrollment_id]
+
+    CSV.generate(headers: true) do |csv|
```
Comment (skelz0r) [inline backend/app/models/enrollment.rb:209]: not sure `headers: true` is needed here.

File: backend/app/models/enrollment.rb [hunk 4]
```diff
@@ -169,6 +171,50 @@ def copy(current_user)
     copied_enrollment
   end
 
+  def team_members_json
```
Comment (skelz0r) [inline backend/app/models/enrollment.rb:174]: Strange to render this field as a json in a CSV, but why not for a first version.

File: backend/config/initializers/cors.rb [hunk 5]
```diff
@@ -11,8 +11,9 @@
     origins(*hosts)
 
     resource "*",
-      methods: %i[get post put patch delete options],
-      headers: %i[Authorization DNT User-Agent X-Requested-With If-Modified-Since Cache-Control Content-Type Range],
+      methods: %i[get post put patch delete options head],
```
Comment (skelz0r) [inline backend/config/initializers/cors.rb:14]: Why ? Need more details in commit message.

**Review by skelz0r — CHANGES_REQUESTED** [review]
There is no test (unit or integration), you should add at least one test on the export and check that the enrollment is a valid CSV (by opening the response's body with `CSV.open(response.body, headers: true)`).

I'm amazed by the ton of code you have to add on the front side btw 😅

File: backend/app/models/enrollment.rb (hunk 4 above)
Comment (rdubigny) [reply backend/app/models/enrollment.rb:174]: agreed

**Pull request merged**